src/common.jl

```diff
@@ -22,6 +22,21 @@ struct HashMismatchError <: Exception
     msg::AbstractString
 end
 
+"""
+Indicate that the feature at hand happens to not be implemented for the given data even
+though it could be. If it is a category mistake to imagine this feature defined on that
+data, use another exception, like `TypeError` or `ArgumentError`.
+"""
+struct UnimplementedError <: Exception
```

Comment:
At risk of nit-picking a good idea unnecessarily, would you mind copying Python's `NotImplementedError`? I write that constantly in my Python work and would like the consistency.

Reply:
Done